Sources/CasePaths/EnumReflection.swift

```diff
@@ -75,6 +86,16 @@ public func extract<Root, Value>(_ embed: @escaping (Value) -> Root) -> (Root) -
     return { _ in nil }
   }
 
+  // https://github.com/pointfreeco/swift-case-paths/issues/40
+  if
+    let wrappedType = metadata.wrappedTypeIfOptional(),
+    wrappedType != Value.self,
+    EnumMetadata(wrappedType) != nil
+  {
+    assertionFailure("embed parameter must not be promoted to return Optional (https://github.com/pointfreeco/swift-case-paths/issues/40)")
```

Comment:
Is it still possible to hit this code path?

Reply:
Seems unlikely to be hit without heroic effort.

Reply:
I wonder if we should maybe not worry about maintaining it and the associated runtime code, then. What do you think?

Reply:
I think I could cook up a scenario involving generics that would call the wrong overload... but I really don't want to. That's what I meant by heroic effort. 😅 I'd personally leave the assertion there, just in case.